pkgs/development/libraries/qt-5/5.6/qtbase/default.nix

```diff
@@ -54,28 +57,41 @@ stdenv.mkDerivation {
       substituteInPlace src/network/kernel/qhostinfo_unix.cpp \
         --replace "@glibc@" "${stdenv.cc.libc.out}"
 
-      substituteInPlace src/plugins/platforms/xcb/qxcbcursor.cpp \
-        --replace "@libXcursor@" "${libXcursor.out}"
-
       substituteInPlace src/network/ssl/qsslsocket_openssl_symbols.cpp \
         --replace "@openssl@" "${openssl.out}"
+    '' + lib.optionalString stdenv.isLinux ''
+      substituteInPlace src/plugins/platforms/xcb/qxcbcursor.cpp \
+        --replace "@libXcursor@" "${libXcursor.out}"
 
       substituteInPlace src/dbus/qdbus_symbols.cpp \
         --replace "@dbus_libs@" "${dbus.lib}"
 
       substituteInPlace \
         src/plugins/platforminputcontexts/compose/generator/qtablegenerator.cpp \
         --replace "@libX11@" "${libX11.out}"
-    ''
-    + lib.optionalString mesaSupported ''
+    '' + lib.optionalString mesaSupported ''
       substituteInPlace \
         src/plugins/platforms/xcb/gl_integrations/xcb_glx/qglxintegration.cpp \
         --replace "@mesa_lib@" "${mesa.out}"
       substituteInPlace mkspecs/common/linux.conf \
         --replace "@mesa_lib@" "${mesa.out}" \
         --replace "@mesa_inc@" "${mesa.dev or mesa}"
+    '' + lib.optionalString stdenv.isDarwin ''
+      sed -i \
+          -e 's|! /usr/bin/xcode-select --print-path >/dev/null 2>&1;|false;|' \
+          -e 's|! /usr/bin/xcrun -find xcrun >/dev/null 2>&1;|false;|' \
```

Comment:
You're looking at a different version of the source than what this commit [downloads](https://github.com/acowley/nixpkgs/blob/c81864fab1d2a0f3896a09ed2a6acd6541ed053c/pkgs/development/libraries/qt-5/5.6/srcs.nix#L38). The line in question is [here](https://github.com/qt/qtbase/blob/v5.6.1/configure#L546). I think this PR has been in review long enough that you're applying it to a newer version than it was written against, so we're not seeing the same things.